src/main/java/bio/terra/cda/app/service/Filter.java

```diff
@@ -332,56 +295,52 @@ public void setCountEndpointQuery() {
-          this.originalReplaceFilterQuery = this.originalReplaceFilterQuery.replace("RIGHT " + fileTableJoinString, "");
-        } else if (this.originalReplaceFilterQuery.contains("FULL " + fileTableJoinString)) {
-          this.originalReplaceFilterQuery = this.originalReplaceFilterQuery.replace("FULL " + fileTableJoinString, "");
-        } else if (this.originalReplaceFilterQuery.contains(fileTableJoinString)) {
-          this.originalReplaceFilterQuery = this.originalReplaceFilterQuery.replace(fileTableJoinString, "");
-        }
-      }
+    String preselect_template = "";
+    if (this.mappingTablePreselect.isEmpty()){
+      preselect_template = "WITH FULLFILTERPRESELECT";
+    } else{
+      preselect_template = "WITH FULLFILTERPRESELECT, FULLMAPPINGPRESELECT";
     }
-    String preselect_template = "WITH file_alias_preselect AS MATERIALIZED (SELECT integer_id_alias FROM file WHERE FILEFILTERS)";
-    this.fileFilterPreselect = replaceKeywords(preselect_template);
-    String where_preselect = file_alias_key + " IN (SELECT integer_id_alias FROM file_alias_preselect)";
-
-    if (this.getNonFileFilters().isEmpty()) {
-      this.fileReplacementFilter = where_preselect;
+    //Build new WHERE filter within paged query
+    String replace_filter_template = "";
+    List<String> joinTableColumnNames = Arrays.stream(this.dataSetInfo
+                    .getTableInfo(this.entityTableName)
+                    .getColumnDefinitions())
+            .sequential().map(ColumnDefinition::getName).collect(Collectors.toList());
+    if (joinTableColumnNames.contains("integer_id_alias")){
+      replace_filter_template = "ENTITYTABLENAME.integer_id_alias IN (UNIONINTERSECT)";
+    } else if (joinTableColumnNames.contains(this.commonAlias)) {
+      replace_filter_template = String.format("ENTITYTABLENAME.%s  IN (UNIONINTERSECT)", this.commonAlias);
     } else {
-      if (this.andFileFilter) {
-        this.fileReplacementFilter = where_preselect + " AND " + this.nonFileFilters;
-      } else {
-        this.fileReplacementFilter = where_preselect + " OR " + this.nonFileFilters;
+      throw new RuntimeException("Unknown column to use for filter");
+    }
+    this.pagedReplacementFilter = replaceKeywords(replace_filter_template);
+
+    //Remove unnecessary joins
+    String originalJoinString = this.originalQuery
+                                .substring(this.originalQuery
+                                      .indexOf(replaceKeywords("FROM ENTITYTABLENAME AS ENTITYTABLENAME")),
+                                                          this.originalQuery.indexOf("WHERE"));
+    String originalSelectString = this.originalQuery
+                                  .substring(0, this.originalQuery
+                                      .indexOf(replaceKeywords("FROM ENTITYTABLENAME AS ENTITYTABLENAME")));
+    List<String> joinList = List.of(originalJoinString.split("(?=(LEFT|INNER|RIGHT|FULL) JOIN)"));
```

Comment:
Consider eating whitespace before join in regex

Reply:
Fixed